pkg/virt-api/webhooks/defaults.go

```diff
@@ -50,9 +50,18 @@ func SetDefaultVirtualMachineInstance(clusterConfig *virtconfig.ClusterConfig, v
 	setDefaultHypervFeatureDependencies(&vmi.Spec)
 	setDefaultCPUArch(clusterConfig, &vmi.Spec)
 	setGuestMemoryStatus(vmi)
+	setGuestCPUTopologyStatus(vmi)
```

Comment:
The idea of an object status is to reflect the current state. Setting that state as dependent **only** on the desired state feels wrong to me.
The request may not even be served (e.g. pod created), but still these status fields are set.

Why would you even need these fields if the status is not dependent on the actual resources?

For future readers, it will be nice to provide the answers in the commit message.
Especially to explain the reason for the change (the reason is now just “bug”).

Reply:
@EdDev Indeed, feels wrong. I would expect it to be set i.e. by the handler, after its received that the domain is running and healthy.